test/sync/src/commonMain/kotlin/io/realm/entities/sync/SyncObjectWithAllTypes.kt

```diff
@@ -0,0 +1,260 @@
+    var longField: Long = 0
+    var booleanField: Boolean = true
+    var doubleField: Double = 0.0
+    var floatField: Float = 0.0.toFloat()
+    var realmInstantField: RealmInstant = RealmInstant.MIN
+    var objectIdField: ObjectId = ObjectId.create()
+    var objectField: SyncObjectWithAllTypes? = null
+
+    // Nullable types
+    var stringNullableField: String? = null
+    var byteNullableField: Byte? = null
+    var charNullableField: Char? = null
+    var shortNullableField: Short? = null
+    var intNullableField: Int? = null
+    var longNullableField: Long? = null
+    var booleanNullableField: Boolean? = null
+    var doubleNullableField: Double? = null
+    var floatNullableField: Float? = null
+    var realmInstantNullableField: RealmInstant? = null
+    var objectIdNullableField: ObjectId? = null
+    var objectNullableField: SyncObjectWithAllTypes? = null
+
+    // RealmLists
+    var stringRealmList: RealmList<String> = realmListOf("hello world")
+    var byteRealmList: RealmList<Byte> = realmListOf(0)
+    var charRealmList: RealmList<Char> = realmListOf(0.toChar())
+    var shortRealmList: RealmList<Short> = realmListOf(0)
+    var intRealmList: RealmList<Int> = realmListOf(0)
+    var longRealmList: RealmList<Long> = realmListOf(0)
+    var booleanRealmList: RealmList<Boolean> = realmListOf(true)
+    var doubleRealmList: RealmList<Double> = realmListOf(0.0)
+    var floatRealmList: RealmList<Float> = realmListOf(0.0.toFloat())
+    var realmInstantRealmList: RealmList<RealmInstant> = realmListOf(RealmInstant.MIN)
+    var objectIdRealmList: RealmList<ObjectId> = realmListOf(ObjectId.create())
+    var objectRealmList: RealmList<SyncObjectWithAllTypes> = realmListOf()
+
+    // Nullable RealmLists of primitive values, not currently supported by Sync
+    // Nullable Object lists, not currently supported by Core
+
+    companion object {
+
+        // Mapping between each Core Field type and functions that can insert data for that type
+        // and also verify the value. This can be used to test objects that has been roundtripped
+        // through Sync.
+        val mapper: Map<TypeDescriptor.CoreFieldType, Pair<FieldDataFactory, FieldValidator>> = createMapper()
+
+        /**
+         * Create mapping for both creating and validating sample data
+         */
+        @Suppress("LongMethod")
+        private fun createMapper(): Map<TypeDescriptor.CoreFieldType, Pair<FieldDataFactory, FieldValidator>> {
+            val map: MutableMap<TypeDescriptor.CoreFieldType, Pair<FieldDataFactory, FieldValidator>> = mutableMapOf()
+            // TODO Problems with accessing TypeDescriptor enums before class is fully initialized
+            //  work-around for now by accessing a property
+            TypeDescriptor.classifiers
+            TypeDescriptor.CoreFieldType.values().forEach { type: TypeDescriptor.CoreFieldType ->
+                map[type] = when (type) {
+                    TypeDescriptor.CoreFieldType.INT -> {
```

Comment:
Not important, but could have put all this in a map with `to` constructors to separate definition from implementation and then just do a generic implementation looking things up ... just sometimes easier to grasp as it is easier to keep an overview on the logic 🤷  
```
TypeDescriptor.CoreFieldType.Int to (
    { obj: SyncObjectWithAllTypes ->
                                obj.intField = 42
                                obj.intNullableField = 42
                                obj.intRealmList = realmListOf(42)
    } to
    { obj: SyncObjectWithAllTypes ->
                                assertEquals(42, obj.intField)
                                assertEquals(42, obj.intNullableField)
                                assertEquals(42, obj.intRealmList.first())
     }
)
```